src/framework/uicomponents/view/validators/doubleinputvalidator.cpp

```diff
@@ -31,67 +31,86 @@ DoubleInputValidator::DoubleInputValidator(QObject* parent)
 
 void DoubleInputValidator::fixup(QString& string) const
 {
-    auto zeros = [](int num)->QString {
-        QString s = QString();
-        for (int i = 0; i < num; i++) {
-            s.append("0");
+    auto removeTrailingZeros = [](QString str)->QString {
+        QString s = str;
```

Comment:
We should pass str by reference and modify it directly rather than creating a copy of it